src/js/components/StageProgress.js

```diff
@@ -0,0 +1,231 @@
+import StageActionButtons from '../components/StageActionButtons';
+import StageLinks from '../components/StageLinks';
+import StringUtil from '../utils/StringUtil';
+
+const METHODS_TO_BIND = ['handleDetailsActionClick'];
+
+class StageProgress extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      nodeListOpen: false
+    };
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentWillMount() {
+    this.props.store.init();
+  }
+
+  componentDidMount() {
+    InstallerStore.setNextStep({
+      enabled: false,
+      label: this.props.nextButtonText,
+      link: null,
+      clickHandler: this.props.nextButtonAction,
+      visible: true
+    });
+  }
+
+  handleDetailsActionClick() {
+    this.setState({nodeListOpen: !this.state.nodeListOpen});
+  }
+
+  handleRetryClick() {
+    this.refs.progressBar.refs.timer.resetTimer();
+    this.props.store.beginStage({retry: true});
+    this.props.store.init();
+  }
+
+  getHeaderIcon(progressState = {}) {
+    if (progressState.completed && progressState.errorCount === 0) {
+      return (
+        <IconCircleCheckmark
+          classNames="icon-padded"
+          key="icon-circle-checkmark" />
+      );
+    }
+
+    if (progressState.completed && progressState.errorCount > 0) {
+      return <IconWarning className="icon-padded" key="icon-warning" />;
+    }
+
+    return this.props.stageIcon;
+  }
+
+  getHeaderContent(progressState = {}) {
+    if (progressState.completed) {
+      if (progressState.failed) {
+        return `${this.props.stateText} Failed`;
+      }
+
+      if (progressState.errorCount > 0) {
+        return `${this.props.stateText} Completed with Errors`;
+      }
+
+      return `${this.props.stateText} Complete`;
+    }
+
+    return this.props.runningText;
+  }
+
+  getProgressBarLabel(progressState = {}) {
+    if (progressState.completed) {
+      return 'Check Complete';
+    }
+
+    let nodeCount = progressState.runningCount;
+    let nodeText = StringUtil.pluralize('Node', nodeCount);
+
+    return (
+      <span>
+        {`Checking ${nodeCount} ${nodeText} `}
+        <span className="progress-bar-label-precentage-complete">
+          ({progressState.percentComplete}% Complete)
+        </span>
+      </span>
+    );
+  }
+
+  render() {
+    let {store} = this.props;
+
+    let completed = store.isCompleted();
+    let detailsAction = 'Show';
+    let errorCount = store.get('errorCount') || 0;
+    let errors = store.get('errorDetails');
+    let failed = store.isFailed();
+    let nodes = store.get('nodes');
+    let runningCount = store.get('runningCount');
+    let successCount = store.get('successCount');
+    let totalHosts = store.get('totalHosts') || 0;
+
+    let percentSuccess = 0;
+    let percentError = 0;
+
+    if (totalHosts !== 0) {
+      percentSuccess = Number((successCount / totalHosts * 100).toFixed(0));
```

Comment:
can we add these to the store? like `getPercentSuccess` or something? you can add it into the `getActionMixin`